multiapi-engine/src/main/java/com/sngular/api/generator/plugin/asyncapi/AsyncApiGenerator.java

```diff
@@ -201,65 +201,70 @@ private JsonNode solveRef(final Path ymlParentPath, final String componentName,
     if (filePath.endsWith("yml") || filePath.endsWith("json")) {
       final ObjectMapper om = new ObjectMapper(new YAMLFactory());
       final var node = om.readTree(file);
-      if (Objects.nonNull(node.findValue(componentName))) {
-        returnNode = node.findValue(componentName);
-        checkReference(node, returnNode, ymlParentPath, totalSchemas);
+      if (Objects.nonNull(node.findValue(path[path.length - 2]).get(path[path.length - 1]))) {
```

Comment:
Are you sure about the length it's always 2 or more? If its ok, but if not I think you should add a exception to avoid losing traceability

Reply:
The development was to include always in the referenced object the path including if it's an schema or a message to avoid problems when a schema and a message has the same name.
So , yes, if the array doens't includes two elements (the package name, and the class name) it's a bug